### PR TITLE
Updates to the notebooks to work with the new registry interface

### PR DESCRIPTION
**Pull request by zoghbi-a**
Doing a fresh PR so it is easier to see the new code changes.

### CODE REVIEW

File: UseCase_III.ipynb [hunk 1]
```diff
@@ -280,11 +280,11 @@
    "source": [
     "bibcode = '1970MmRAS..73..125E' # Eichhorn\n",
     "idx=-1\n",
-    "for s in tap_services.to_table():\n",
+    "for s in tap_services:\n",
     "    idx+=1\n",
     "    if bibcode in s['source_value']:\n",
     "        myidx=idx\n",
-    "        print(f\"{s['short_name']}, {s['source_value']}, {s['access_url']}\")\n",
+    "        print(f\"{s.short_name}, {s.source_value}, {s.access_url}\")\n",
```
[the remark this answers is not in the document]
Comment (zoghbi-a) [reply UseCase_III.ipynb:287]: That's right. It doesn't work with the current pyvo version. The changes are for the new pyvo registry interface, and unfortunately, some changes in the new registry interface are not compatible with older pyvo version.

One of the issues, is  that `access_url` is not always available in the new interface. There is `access_urls` that gives a list of urls. `to_table` doesn't pass the `access_url(s)` to the table, so to access it, we have to use the service object directly (without `to_table`).

`s.access_url` defaults to the first element in `access_urls`.  I think it has to do with multi-capability resources.
Here is part of the [discussion in the pyvo PR](https://github.com/astropy/pyvo/pull/289#discussion_r785382094).

File: UseCase_III.ipynb [hunk 2]
```diff
@@ -322,7 +322,7 @@
     "    if 'Eichhorn' in names[0]: \n",
     "        print(\"For %s: \" %record.short_name)\n",
     "        print(\"     Reference URL: %s\" %record.reference_url)\n",
-    "        print(\"     Access URL: %s\" %record.access_url)\n",
+    "        print(\"     Access URL: %s\" %record['access_urls'][0])\n",
```
[the remark this answers is not in the document]
Comment (zoghbi-a) [reply UseCase_III.ipynb:325]: The new interface has `access_urls`. See my comment above.

File: UseCase_III.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (trjaffe) [reply UseCase_III.ipynb:287]: So to be clear, this PR on navo-workshop goes with  Markus' PR on pyvo here: 
 https://github.com/msdemlei/pyvo/tree/add-discoverdata Right?  So they should be merged at the same time.  Are we planning on this before the AAS?  Sorry I've lost track.  I thought we were going to update the notebooks in a way that would work with both, but apparently we cannot.

[the remark this answers is not in the document]
Comment (bsipocz) [reply UseCase_III.ipynb:287]: Merging should be done on pyvo prior to this PR. Then we can still roll back the decision and not go ahead with an actual release/tinker more on pyvo dev based on the experience we'll have in this PR.

I'll push the test matrix PR in the next few hours, and will cherry-pick this commit as being the test case to show how we fare with different version.

**Comment by bsipocz** [comment]
I've rebased this on top of the refactored CI to provide some tests for the upcoming PyVO release.

**Comment by bsipocz** [comment]
So, here all jobs are expected to fail that are not using the development version of pyvo (<v1.4). 

However, unfortunately I see a lot of failures with the dev version testing, too, full log is here: https://github.com/NASA-NAVO/navo-workshop/actions/runs/3094751660/jobs/5008455576

None of the looks specifically PyVO related, but @tomdonaldson, could you please have a look at this before you push the release, as they seem to be complaining about unavailable STScI URLs, which at the end of the may or may not have a pyvo origin.

**Comment by tomdonaldson** [comment]
I got the tests working by changing the workaround for the SDSS SIA service to specify `format=None` instead of `format=''`.  This was needed due to a change in how `pyvo` decided when to set that parameter (`if format:` became `if format is not None:`).

I also attempted to update the minimum version of `pyvo` to `1.4`, but could easily have missed something.  This should be ready for re-review.

I did not yet look into why the skipped test is skipped.  I expect it's obvious but I need to step away for a bit.

**Comment by tomdonaldson** [comment]
> I did not yet look into why the skipped test is skipped. I expect it's obvious but I need to step away for a bit.

Yes it looks like 'Publish HTML' only runs on pushes to main.

**Review by bsipocz — APPROVED** [review]
LGTM

**Comment by bsipocz** [comment]
Thanks @zoghbi-a and @tomdonaldson!

**Pull request merged**